freezefrog/__init__.py

```diff
@@ -28,100 +65,124 @@ def __instancecheck__(self, obj):
 
 class FakeDateTime(with_metaclass(FakeDateTimeMeta, real_datetime)):
     """
-    A datetime replacement that lets you set utcnow() using set_utcnow().
+    A `datetime` mock class.
 
-    The clock starts ticking after calling set_utcnow().
+    Set the start datetime and the system timezone with `start(dt, tz)`. The
+    clock starts ticking after calling `start`.
     """
-    tz_delta = None
 
-    @classmethod
-    def utcnow(cls, *args, **kwargs):
-        if not hasattr(cls, 'dt'):
-            raise NotImplementedError(
-                'use {}.set_utcnow(datetime) first'.format(cls.__name__)
-            )
-        return cls._utcnow()
+    dt = None
+    tz = None
+    _start = None
 
     @classmethod
-    def set_utcnow(cls, dt):
+    def start(cls, dt, tz, tick, fold):
         cls.dt = dt
+        cls.tz = tz
+        cls._start = time.monotonic() if tick else None
+
+        if is_pytz(tz):
+            if not PYTZ_SUPPORT:
+                raise Exception(
+                    "pytz not supported, please install pytz first"
+                )
+
+            cls.is_dst = get_pytz_dst_from_fold(dt, tz, fold)
+            cls.now = cls.now_with_pytz
+            cls.utcnow = cls.utcnow_with_pytz
+        else:
+            cls.fold = fold
+            cls.now = cls.now_with_datetime_tz
+            cls.utcnow = cls.utcnow_with_datetime_tz
 
     @classmethod
-    def set_tz_delta(cls, tz_delta):
-        cls.tz_delta = tz_delta
+    def time_since_start(cls):
```

Comment:
Since these are exposed on the mocked `datetime` class, should we prefix all custom methods that are not intended to be used by the end user with underscore?

Reply:
Done in https://github.com/closeio/freezefrog/pull/13/commits/f6fd480f3cab17a523cef23b74502e5904ec8098.